src/main/java/data/Vendor.java

```diff
@@ -64,4 +64,24 @@ public void addAccount(Account account) {
   public ArrayList<Account> getAccounts() {
     return accountList;
   }
+
+  /** Return a JSON String representing a billing config's content. */
+  public String buildJsonConfig() {
+    ArrayList<Account> accounts = getAccounts();
+    String config = toJson();
+
+    for(int i = 0; i < accounts.size(); i++) {
+      config += accounts.get(i).toJson();
+    }
+    config += "]}";
+
+    return config;
+  }
+
+  /** Return a JSON String representation of Vendor fields. */
+  public String toJson() {
+    return String.format("{\"legacy_vendor_id\":%s," +
+      "\"next_gen_vendor_id\":%d,\"accounts\":[", getLegacyVendorID(), 
+      getNextGenVendorID());
+  }
```

Comment:
I think it would be good to inline this method in the one above (`String config = String.format(...);`). It is a little strange to see only the closing "]" of the list added in the above function.

Either that, or make this private and move the account loop here.

Reply:
I moved the toJson stuff directly in the buildJsonConfig method.